pisa-proxy/parser/mysql/src/grammar.y

```diff
@@ -6262,6 +6270,290 @@ default_encryption -> CreateDatabaseOption:
+	      	s.push_str(&$1.0);
+	      	s.push('.');
+	      	s.push_str(&$3.0);
+	      	s
+          }
+        | ident
+          {
+          	$1.0
+          }
+        ;
+
+trg_action_time -> TrgActionTime:
+            'BEFORE'
+            { TrgActionTime::Before }
+          | 'AFTER'
+            { TrgActionTime::After }
+          ;
+
+trg_event -> TrgEvent:
+            'INSERT'
+            { TrgEvent::Insert }
+          | 'UPDATE'
+            { TrgEvent::Update }
+          | 'DELETE'
+            { TrgEvent::Delete }
+          ;
+
+trigger_follows_precedes_clause -> TriggerFollowsPrecedesClause:
+            /* empty */
+            {
+            	TriggerFollowsPrecedesClause{
+            		ordering_clause: TrgActionOrder::None,
+            		anchor_trigger_name: None,
+            	}
+            }
+          |
+            trigger_action_order ident_or_text
+            {
+              	TriggerFollowsPrecedesClause{
+                          		ordering_clause: $1,
+                          		anchor_trigger_name: Some($2),
+                          	}
+            }
+          ;
+
+trigger_action_order -> TrgActionOrder:
+            'FOLLOWS'
+            { TrgActionOrder::Follows }
+          | 'PRECEDES'
+            { TrgActionOrder::Precedes }
+          ;
+
+sp_proc_stmt -> Option<String>:
+         /* TODO */
+         { None }
+        ;
+
+sp_tail -> SpTail:
+          'PROCEDURE'
+          sp_name
+          '('
+          sp_pdparam_list
+          ')'
+          sp_c_chistics
+          sp_proc_stmt
+          {
+          	SpTail{
+          		span: $span,
+          		sp_name: $2,
+          		sp_pdparam_list: $4,
+          		sp_c_chistics: $6,
+          		sp_proc_stmt: $7,
+          	}
+          }
+        ;
+
+sp_pdparam_list -> Vec<SpPdparam>:
+          /* Empty */ { vec![] }
+        | sp_pdparams { $1 }
+        ;
+
+sp_pdparams -> Vec<SpPdparam>:
+          sp_pdparams ',' sp_pdparam
+          {
+          	$1.push($3);
+          	$1
+          }
+        | sp_pdparam
+        {
+        	vec![$1]
+        }
+        ;
+
+sp_pdparam -> SpPdparam:
+          sp_opt_inout ident type opt_collate
+          {
+            	SpPdparam{
+            		sp_opt_inout: $1,
+            		ident: $2.0,
+            		sp_type: $3,
+            		opt_collate: $4,
+            	}
+          }
+        ;
+
+sp_opt_inout -> SpOptInout:
+          /* Empty */ { SpOptInout::In }
+        | 'IN'      { SpOptInout::In }
+        | 'OUT'     { SpOptInout::Out }
+        | 'INOUT'   { SpOptInout::Inout }
+        ;
+
+sp_c_chistics -> Vec<String>:
+          /* Empty */ { vec![] }
+        | sp_c_chistics sp_c_chistic
+        {
+        	$1.push($2);
+        	$1
+        }
+        ;
+
+sp_c_chistic -> String:
+          sp_chistic            { $1 }
+        | 'DETERMINISTIC'     { String::from("DETERMINISTIC") }
+        | not 'DETERMINISTIC' { String::from("NOT DETERMINISTIC") }
+        ;
+
+sp_chistic -> String:
```

Comment:
This rule should also return a `Enum` type.

Reply:
Ok, Thank you , Done.